### DISCUSSION TITLE
using opacity for watermark

### DISCUSSION THREAD

**Discussion by Carl873** [ImageMagick/ImageMagick#5781] [Help] [answered]
Hi, I add a watermark that is an image, onto another image, like so:
```
  magick `
  $sourceImage `
  -strip `
  $watermark `
  -gravity southeast `
  -composite  `
  $resultImage
```
and this works, but when I try to make the watermark have less opacity, 
```
  magick `
  $sourceImage `
  -strip `
  $watermark `
  -gravity southeast `
  -dissolve 50% `
  -composite  `
  $resultImage
```
I run into an error:
```
magick.exe: unrecognized option `-dissolve' at CLI arg 6 @ fatal/magick-cli.c/ProcessCommandOptions/659.`
```
I'm on Windows, using PowerShell and magick.exe, 
```
version: ImageMagick 7.1.0-52 Q16 x64 04ee6ce:20221106
```
$sourceImage is jpg, $watermark is png

How can I make this work? I should add that some examples of magick syntax that I find, contain '(', ')' but I cannot make these work in PS. Do I need brackets for this?

**Comment by fmw42** [comment, marked as answer]
On Linux or Mac

```
 magick \
  $sourceImage \
  -strip \
  $watermark \
  -gravity southeast \
  -define compose:args=50,50 \
  -compose dissolve \
  -composite  \
  $resultImage
```

see https://imagemagick.org/Usage/compose/#dissolve

**Reply by Carl873** [reply]
Thank you! It works! But on a side note, I found that `args=50,50` does not give me the expected results (that the watermark has 50% opacity). Only `args=50` or `args=50,100` does. But the most important thing is that it accepts 'dissolve' now. :)

**Reply by fmw42** [reply]
50,100 is good. I was thinking blending argument that need to add to 100, but you wanted dissolve, which do not have to add to 100.  But choose any value that works for what you want.
